src/test/suite/connectionController.test.ts

```diff
@@ -915,4 +924,328 @@ suite('Connection Controller Test Suite', function () {
+      `Expected connections to be 1 found ${connections.length}`
+    );
+
+    const mockMigratePreviouslySavedConnection: any = sinon.fake();
+
+    sinon.replace(
+      testConnectionController,
+      '_migratePreviouslySavedConnection',
+      mockMigratePreviouslySavedConnection
+    );
+
+    const newSavedConnectionInfoWithSecrets = await testConnectionController._getConnectionInfoWithSecrets(connections[0]);
+
+    assert.deepStrictEqual(
+      newSavedConnectionInfoWithSecrets,
+      connectionInfo
+    );
+    assert(
+      mockMigratePreviouslySavedConnection.called === false,
+      'Expected _migratePreviouslySavedConnection to not be called.'
+    );
+  });
+
+  test('addNewConnectionStringAndConnect saves connection without secrets to the global store', async () => {
+    const mockConnect: any = sinon.fake.resolves({ successfullyConnected: true });
+
+    sinon.replace(
+      testConnectionController,
+      '_connect',
+      mockConnect
+    );
+
+    await vscode.workspace
+      .getConfiguration('mdb.connectionSaving')
+      .update('defaultConnectionSavingLocation', DefaultSavingLocations.Global);
+    await testConnectionController.addNewConnectionStringAndConnect(
+      TEST_DATABASE_URI_USER
+    );
+
+    const workspaceStoreConnections = testConnectionController._storageController.get(
+      StorageVariables.GLOBAL_SAVED_CONNECTIONS
+    );
+
+    assert(
+      workspaceStoreConnections,
+      `Expected workspace store has onnections found ${workspaceStoreConnections}`
+    );
+
+    const connections = Object.values(workspaceStoreConnections);
+
+    assert(
+      connections.length === 1,
+      `Expected connections to be 1 found ${connections.length}`
+    );
+    assert(
+      connections[0].connectionOptions?.connectionString.includes(TEST_USER_USERNAME) === true,
+      `Expected a connection on disc includes username found ${connections[0].connectionOptions?.connectionString}`
+    );
+    assert(
+      connections[0].connectionOptions?.connectionString.includes(TEST_USER_PASSWORD) === false,
+      `Expected a connection on disc does not include password found ${connections[0].connectionOptions?.connectionString}`
+    );
+    assert(
+      connections[0].connectionOptions?.connectionString.includes('appname=mongodb-vscode+0.0.0-dev.0') === true,
+      `Expected a connection on disc has appname found ${connections[0].connectionOptions?.connectionString}`
+    );
+    assert(
+      testConnectionController._connections[connections[0].id].connectionOptions?.connectionString.includes(TEST_USER_PASSWORD) === true,
+      `Expected a connection in memory includes password found ${connections[0].connectionOptions?.connectionString}`
+    );
+    assert(
+      testConnectionController._connections[connections[0].id].name === 'localhost:27018',
+      `Expected a connection has the 'localhost:27018' name found ${testConnectionController._connections[connections[0].id].name}`
+    );
```

Comment:
nit: It would be a bit less verbose if we only add messages for assertions when they provide extra context, e.g. here they don’t and you could just do:

```suggestion
    assert.strictEqual(
      testConnectionController._connections[connections[0].id].name,
      'localhost:27018'
    );
```

and get a similar (or better) error message